src/components/auth/flow.tsx

```diff
@@ -207,7 +217,7 @@ export function handleFlowError<S>(
         return
       case 'session_already_available':
         // User is already signed in, let's redirect them home!
-        await router.push('/auth/login-check')
+        await router.push('/')
```

Comment:
Bonus points:
- add a toast message like: "Welcome back – you are already logged in"
- maybe try redirecting to last page first